src/relay/transforms/partition_graph.cc

```diff
@@ -456,18 +370,111 @@ class Partitioner : public ExprMutator {
   }
 
   /*!
-   * \brief Get the index of the return(output);
-   * this is to be used as tuplegetitem idx
+   * \brief This function is called first time that we encounter a compiler_end
+   * node to create the function for the subgraph.
    */
-  int GetRetIdx(AnnotatedRegion sg, const Expr& arg) {
-    int idx = 0;
-    for (auto arg_ : sg->GetOutputs()) {
-      if (arg == arg_) {
-        return idx;
+  void CreateFunction(AnnotatedRegion region, const CallNode* call) {
```

Comment:
Out of curiosity, why is it better to handle the fix here than in the `AnnotateTarget` pass? It seems that it would work if `AnnotateTarget` just didn't generate the extra `compiler_end` annotations.

Reply:
Actually, the way the MergeCompilerRegions algorithm works, the annotations need to adhere to the rule that they a single input and single output. (See step 3 merging in this [RFC](https://discuss.tvm.ai/t/rfc-byoc-an-extended-graph-partitioning-flow/6028) ).

Reply:
So the fan-out nodes need to have a single annotation per output, because when MCR works, it may be the case one fan-out path be merged and other output remains as a output.

Reply:
That's the point. `AnnotateTarget` does generate extra `compiler_end` annotations. To keep the implementation clear and simple, we let each node insert `compiler_end` to its arguments, resulting in some extra `compiler_end` annotations.

Reply:
I see, thanks for the explanation!